internal/store/db/mongo/store.go

```diff
@@ -16,37 +16,254 @@
 package mongo
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/contracts"
 	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/db"
 	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/db/interfaces"
-	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/models"
+	"github.com/edgexfoundry/app-functions-sdk-go/internal/store/db/mongo/models"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+// Client provides a wrapper for Mongo's Client type
+type Client struct {
+	Timeout time.Duration
+	Client  *mongo.Database
+	// unexported Client for Disconnect
+	client *mongo.Client
+}
+
+const mongoCollection = "store"
+
 // Store persists a stored object to the data store.
-func Store(o models.StoredObject) error {
-	return nil
+func (c Client) Store(o contracts.StoredObject) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout)
+	defer cancel()
+
+	objID, uuid, err := models.FromContractId(o.ID)
+	if err != nil {
+		return "", err
+	}
+	var doc bson.M
+
+	if objID == primitive.NilObjectID {
+		// determine if this object already exists in the DB
+		filter := bson.M{"uuid": uuid}
+		result := c.Client.Collection(mongoCollection).FindOne(ctx, filter)
+
+		var m models.StoredObject
+		_ = result.Decode(&m)
+
+		// if the result of the lookup is any object other than the empty, it exists
+		if !reflect.DeepEqual(m, models.StoredObject{}) {
+			return "", errors.New("object exists in database")
+		}
+
+		doc = bson.M{
+			"uuid":             uuid,
```

Comment:
Don't want to create the new objID ? or will Mongo create it?

Reply:
Mongo will create one.